### DISCUSSION TITLE
Surplus energy for therminal battery

### DISCUSSION THREAD

**Discussion by LaurensVanAcker** [davidusb-geek/emhass#277] [Q&A]
Is there a method to incorporate a surplus energy device?

I have a solar boiler running in series with a gas heater, rendering it unnecessary to maintain a minimum temperature as it's post-heated by gas. However, I've installed a variable resistor (100 - 3000W) to convert surplus energy into usable terminal energy.

Is there a way to integrate this device? For instance, if energy prices are negative or near zero, and we anticipate reaching full battery capacity later, could we activate it to convert solar energy into stored therminal power?

**Comment by davidusb-geek** [comment]
Ok in the other answer #282 I stated that a good solution was to add support for PV curtailment. But your case is a bit different indeed and I see your need for this feature. This is not supported for now but it could be a good future feature.

However there are solutions using the power of automations in HA.
Here is a proposition:
1- Add a dummy deferrable load in EMHASS to model your surplus thermal load. But define it with `total_def_hours=0`.
2- Launch EMHASS optimization and publish the results to HA
3- In HA setup an automation that check and computes the amount of energy in Wh that is injected to the grid during negative price periods. We could further discuss on how to achieve this but we have all the needed information on the published sensors attributes to do it.
4- Then call EMHASS optimization again but now setting `total_def_hours` and the nominal power for your dummy load accordingly to the computed energy in Wh from the previous step. In your case setup this accordingly with your resistor device power rating.

The result of this last optimization should schedule your dummy load to avoid injection to the grid on negative values.
If you are willing to test this approach I will gladly help of needed on settings the automations.
It it works it could be nice to document it and share it on the EMHASS documentation.
